### ISSUE TITLE
Feature: logrotate configs

### ISSUE THREAD

**Issue by klamontagne** [sous-chefs/mysql#294]
The mysql cookbook should add a logrotate file for each instance.

The logrotate file could be removed for the disabled "default" server, as well.

Currently, the deb package for mysql server leaves the following (taken from the service55-single-debian-74 kitchen) in /etc/logrotate.d/mysql-server : 

```
# - I put everything in one block and added sharedscripts, so that mysql gets
#   flush-logs'd only once.
#   Else the binary logs would automatically increase by n times every day.
# - The error log is obsolete, messages go to syslog now.
/var/log/mysql.log /var/log/mysql/mysql.log /var/log/mysql/mysql-slow.log {
  daily
  rotate 7
  missingok
  create 640 mysql adm
  compress
  sharedscripts
  postrotate
    test -x /usr/bin/mysqladmin || exit 0
    # If this fails, check debian.conf!
    MYADMIN="/usr/bin/mysqladmin --defaults-file=/etc/mysql/debian.cnf"
    if [ -z "`$MYADMIN ping 2>/dev/null`" ]; then
      # Really no mysqld or rather a missing debian-sys-maint user?
      # If this occurs and is not a error please report a bug.
      #if ps cax | grep -q mysqld; then
      if killall -q -s0 -umysql mysqld; then
        exit 1
      fi
    else
      $MYADMIN flush-logs
    fi
  endscript
}
```

The postrotate script fails and cron.daily throws this error every day after the first:

```
/etc/cron.daily/logrotate:
error: error running shared postrotate script for '/var/log/mysql.log /var/log/mysql/mysql.log  /var/log /mysql/mysql-slow.log /var/log/mysql/error.log '
run-parts: /etc/cron.daily/logrotate exited with return code 1
```

**Comment by ambientsound** [comment]
Much appreciated if this feature is implemented.

**Comment by someara** [comment]
I'm not sure if this belongs in the mysql_service resource internal
implementation...

A key feature of the primitive is that it *behaves* exactly the same across
every platform it supports. If we were to implement this, it would need to
behave the same way across

debian-7
debian-8
ubuntu-10.04
ubuntu-12.04
ubuntu-14.04
centos-5
centos-6
centos-7
omnios
smartos
windows2012r2

That being said... I'm *totally* down to see this feature. The quickest
path to happy is probably to imlement it externally, using the logrotate
cookbook until all the bugs are shaken out.

-s

**Comment by klamontagne** [comment]
I think I conflated two issues: the feature and the bug with logrotate and the disabled "packaged" instance.

**Comment by someara** [comment]
Oh right.

The provider should *definitely* cleanup any broken logs being generated by
vendor package configurations.

Good call.

I even labeled the issue as such!

-s

**Comment by nkadel-skyhook** [comment]
I'll beg to disagree. Touching logs of a service you didn't install or configure and are not managing is eliminating the very evidence needed to compare the old and new setup results.

**Comment by someara** [comment]
We are managing it.

We're managing it so that it's state is [:stop, :disable].

The issue is that the Debian packaging *also* implements a logrotate cron
that assumes [:start, :enable].

By not configuring that as well, we're effectively only "half managing" the
mysql_service.

-s

On Thu, Apr 16, 2015 at 1:14 PM, Nico Kadel-Garcia <notifications@github.com

**Comment by flatrocks** [comment]
We needed a functional workaround for some production servers (Ubuntu 14.04) so I created [flatrocks/mysql_logrotate](https://github.com/flatrocks/cookbook-mysql_logrotate) cookbook.

It includes a resource to implement log rotation for a mysql_service instance, essentially copying the setup that worked with older versions of the mysql cookbook, with the help of the logrotate cookbook.  Log rotation, log-flushing, etc, should only affect the designated mysql_service resource.  It overcomes the lack of a mysql debian system user account by creating a minimum-privilege user to flush the logs.

That left us with the error, originally noted by @klamontagne.  We solved this by simply adding this to a recipe:
``` ruby
logrotate_app 'mysql-server' do
  enable false
end
```
Hoping someone finds it useful.  It feels rather hackish, but now we've got working log rotation without the error messages.
